### ISSUE TITLE
"delete" key maps to "shift p" on Japanese keyboard

### ISSUE THREAD

**Issue by michael-keyes** [gokcehan/lf#486]
"delete" key maps to "shift p" on Japanese keyboard. Tested on 3 different machines

**Comment by Cyberdr3am** [comment]
Hi, on french keyboard when i use "delete" lf think i press "shift p"      ==>    "P"

**Comment by gokcehan** [comment]
@michael-keyes @Cyberdr3am Do you not get the same behavior with english layout? What terminal are you using? If your terminal sends `P` when `delete` is pressed, there is nothing much we can do. There may be a setting in your terminal about what key or escape sequence to send when `backspace` or `delete` is used.

**Comment by Cyberdr3am** [comment]
@gokcehan

Thanks for your help.
I use st all the time and no problem with the older release but the last one broke something.
Urxvt and xterm run delete key without problem.
This is my build of st : https://github.com/LukeSmithxyz/st

**Comment by michael-keyes** [comment]
@gokcehan 
I use st with limited patches. The previous version of lf had no issues with the same st version and build I currently use

**Comment by gokcehan** [comment]
@michael-keyes @Cyberdr3am We have changed our terminal ui library in r17 from termbox to tcell and they may handle things differently. I don't know much about st. Is [delkey](https://st.suckless.org/patches/delkey/) patch relevant for this?

**Comment by Cyberdr3am** [comment]
I don't have any problem with other terminal file manager with st.
The patch delkey is great when the terminal don't know this key but all file manager in st work with this key.
I can change my bind to other key if this fix the problem

**Comment by ivanp7** [comment]
On my setup (LukeSmithxyz/st, russian keyboard) <delete> is also interpreted as <P>, but "unknown mapping: <a-[>" is printed into the status line.

**Comment by kmwenja** [comment]
I have the same issue (on a normal keyboard) with st (no delkey patch). These tips here helped: https://github.com/chrisallenlane/st/blob/2ee24fc567bb39bb634c871c2614aaf2a6252c5d/FAQ#L36-L89

To repro (in bash):
```bash
# run lf first to prove that Del key maps to shift p
lf  # try to delete something

# using tput
tput smkx

# run lf this time to see if Del key works properly
lf
```

I won't admit to understanding the fundamental cause (the issue with the keypad Del) but reading up on st's terminfo was, at the very least, educational.

**Comment by ivanp7** [comment]
@gokcehan Tried delkey patch, issue is not affected.

**Comment by OliverLew** [comment]
Could this be a `st` issue? I found a lot of mailing list threads about delete key malfunctioning, not specific to tui based on tcell:
https://lists.suckless.org/dev/1212/13800.html
https://lists.suckless.org/dev/1212/13848.html (this is a dev responding)
https://lists.suckless.org/dev/1301/13963.html
https://lists.suckless.org/dev/1302/14464.html
https://lists.suckless.org/dev/1304/15114.html
https://lists.suckless.org/dev/1404/20777.html
https://lists.suckless.org/dev/2002/33795.html
Anyone know these stuff and figure out what the problem is?

Also, like what is described in
https://lists.suckless.org/dev/1212/13802.html,
if anyone tried 'insert' key in `st` and found out you just did `4h` (4 directories up), then we are having the same issue :cry:

**Comment by gokcehan** [comment]
I have looked into this. It seems that tcell does not send the enter keypad terminfo on startup. What's more is that it was sending exit keypad on quit so even if you set `tput smkx` before running `lf` it would still reset on quit. I think this was an oversight so I have added this to the patch that I sent for our other issue #586 in PR https://github.com/gdamore/tcell/pull/439 . Let's see how it goes.

In the meantime, I have switched to our local fork so this should work as intended in our master branch now.
